File: mobile/src/api/journey.ts
import { apiRequest } from "./client";

export type JourneyCategory = {
  category: string;
  display_name: string;
  resolution_id: string;
  resolution_title: string;
  total_tasks: number;
  completed_tasks: number;
};

export type DailyJourneyResponse = {
  user_id: string;
  categories: JourneyCategory[];
  request_id: string;
};

export async function fetchDailyJourney(userId: string): Promise<DailyJourneyResponse> {
  const { data } = await apiRequest<DailyJourneyResponse>(`/journey/daily?user_id=${userId}`);
  return data;
}

File: backend/app/api/schemas/journey.py
"""Pydantic schemas for the daily journey widget."""
from __future__ import annotations

from typing import List
from uuid import UUID

from pydantic import BaseModel


class JourneyCategoryPayload(BaseModel):
    category: str
    display_name: str
    resolution_title: str
    total_tasks: int
    completed_tasks: int


class DailyJourneyResponse(BaseModel):
    user_id: UUID
    categories: List[JourneyCategoryPayload]
    request_id: str
